provider/cluster_resource_rosa_test.go

```diff
@@ -0,0 +1,304 @@
+		    name           = "my-cluster"
+			product		   = "rosa"
+		    cloud_provider = "aws"			
+		    cloud_region   = "us-west-1"
+		  }
+		`)
+		Expect(terraform.Apply()).To(BeZero())
+	})
+
+	It("Creates cluster with http proxy", func() {
+		// Prepare the server:
+		server.AppendHandlers(
+			CombineHandlers(
+				VerifyRequest(http.MethodPost, "/api/clusters_mgmt/v1/clusters"),
+				VerifyJQ(`.name`, "my-cluster"),
+				VerifyJQ(`.cloud_provider.id`, "aws"),
+				VerifyJQ(`.region.id`, "us-west-1"),
+				VerifyJQ(`.product.id`, "rosa"),
+				VerifyJQ(`.proxy.http_proxy`, "http://proxy.com"),
+				VerifyJQ(`.proxy.https_proxy`, "http://proxy.com"),
+				RespondWithPatchedJSON(http.StatusOK, template, `[
+					{
+					  "op": "add",
+					  "path": "/proxy",
+					  "value": {						  
+						  "http_proxy" : "http://proxy.com",
+						  "https_proxy" : "http://proxy.com"
+					  }
+					}]`),
+			),
+		)
+
+		// Run the apply command:
+		terraform.Source(`
+		  resource "ocm_cluster" "my_cluster" {
+		    name           = "my-cluster"
+			product		   = "rosa"
+		    cloud_provider = "aws"			
+		    cloud_region   = "us-west-1"
+			proxy = {
+				http_proxy = "http://proxy.com",
+				https_proxy = "http://proxy.com",
+			}			
+		  }
+		`)
+		Expect(terraform.Apply()).To(BeZero())
+	})
+
+	It("Creates cluster with aws subnet ids & private link", func() {
+		// Prepare the server:
+		server.AppendHandlers(
+			CombineHandlers(
+				VerifyRequest(http.MethodPost, "/api/clusters_mgmt/v1/clusters"),
+				VerifyJQ(`.name`, "my-cluster"),
+				VerifyJQ(`.cloud_provider.id`, "aws"),
+				VerifyJQ(`.region.id`, "us-west-1"),
+				VerifyJQ(`.product.id`, "rosa"),
+				VerifyJQ(`.aws.subnet_ids.[0]`, "id1"),
+				VerifyJQ(`.aws.private_link`, true),
+				VerifyJQ(`.nodes.availability_zones.[0]`, "az1"),
+				VerifyJQ(`.api.listening`, "internal"),
+				RespondWithPatchedJSON(http.StatusOK, template, `[
+					{
+						"op": "add",
+						"path": "/availability_zones",
+						"value": ["az1", "az2", "az3"]
+					},					
+					{
+					  "op": "add",
+					  "path": "/aws",
+					  "value": {
+						  "private_link": true,
+						  "subnet_ids": ["id1", "id2", "id3"]
+					  }
+					}]`),
+			),
+		)
+
+		// Run the apply command:
+		terraform.Source(`
+		  resource "ocm_cluster" "my_cluster" {
+		    name           = "my-cluster"
+			product		   = "rosa"
+		    cloud_provider = "aws"			
+		    cloud_region   = "us-west-1"
+			availability_zones = ["az1","az2","az3"]
+			aws_private_link = true
+			aws_subnet_ids = [
+				"id1", "id2", "id3"
+			]
+		  }
+		`)
+		Expect(terraform.Apply()).To(BeZero())
+	})
+
+	It("Creates rosa sts cluster", func() {
+		// Prepare the server:
+		server.AppendHandlers(
+			CombineHandlers(
+				VerifyRequest(http.MethodPost, "/api/clusters_mgmt/v1/clusters"),
+				VerifyJQ(`.name`, "my-cluster"),
+				VerifyJQ(`.cloud_provider.id`, "aws"),
+				VerifyJQ(`.region.id`, "us-west-1"),
+				VerifyJQ(`.product.id`, "rosa"),
```

Comment:
How do we send the operator role prefix ?

Reply:
@pvasant  here is an example. https://github.com/rh-mobb/terraform-provider-ocm/blob/example/example.tf

Reply:
@pvasant  Waiting state is because rosa/OCM API problem. It has a circular dependency, I have to put wait as **false**. The reason is that OCM "create cluster" needs to generate an ID but it is always, then the role can be created/tagged appropriately. And OCM polls the backend resume until the roles are created. I talked with @vkareh as well on the API problem. CC @paulczar

Reply:
@pvasant  anyway we can get this merge?